gui/dialogs/dialogoptimize.cpp

```diff
@@ -221,3 +299,85 @@ void DialogOptimize::finalizePlots() {
     ui->btnStartStop->setText("Start");
   }
 }
+
+void DialogOptimize::updateTargetImage() {
+  // this handles the distinction between 2D and 3D visualizations when setting
+  // image data for the image that displays the target values
+  // and handles the z-slider for the 2D visualization
+  auto variable = ui->cmbTarget->currentIndex();
+  if (m_opt == nullptr || variable < 0 || variable >= ui->cmbTarget->count()) {
+    return;
+  }
+
+  SPDLOG_DEBUG(" Updating target image for variable {}", variable);
+
+  auto img = m_opt->getTargetImage(variable);
+  SPDLOG_DEBUG(" data: {}, empty: {}", img.volume().depth(), img.empty());
+
+  if (vizMode == VizMode::_2D) {
+    SPDLOG_INFO("Updating target image for 2D visualization ");
+    // all the z-axis stuff is only necessary for 2D visualization
+    auto z = ui->zaxisValues->value();
+    ui->lblTarget->setImage(img);
+    ui->zaxisValues->setMinimum(0);
+    ui->zaxisValues->setMaximum(ui->lblTarget->getImage().volume().depth());
+    ui->zaxisValues->setValue(z);
+    ui->lblTarget->setZSlider(ui->zaxisValues);
+    ui->lblTarget->setZIndex(z);
+    ui->lblTarget->updateGeometry();
+
+  } else {
+    SPDLOG_DEBUG("Updating target image for 3D visualization: 2D visible? {}, "
+                 "3d visible? {}",
+                 ui->lblTarget->isVisible(), ui->lblTarget3D->isVisible());
+    ui->lblTarget3D->setImage(img);
+    ui->lblTarget3D->updateGeometry();
+  }
+}
+
+void DialogOptimize::updateResultImage() {
+  // this handles the distinction between 2D and 3D visualizations
+
+  if (m_opt == nullptr) {
+    return;
+  }
+
+  if (vizMode == VizMode::_2D) {
+    auto z = ui->zaxisValues->value();
+    ui->lblResult->setImage(m_opt->getCurrentBestResultImage());
+    ui->lblResult->setZSlider(ui->zaxisValues);
+    ui->lblResult->setZIndex(z);
+    ui->lblResult->updateGeometry();
```

Comment:
can this be removed?
```suggestion
    ui->lblResult->setImage(m_opt->getCurrentBestResultImage());
```

Reply:
the updateGeometry part isn't strictly necessary because the problem it solved was fixed in the ui layout itself, but if I remove the zslider stuff the z-slider looses its functionality upon change from difference mode to values and back it seems.